src/regtech_cleanup_api/entities/repos/institution_repo.py

```diff
@@ -6,28 +7,38 @@
 )
 from sqlalchemy.orm import Session
 from sqlalchemy import text
+from regtech_cleanup_api.entities.repos import repo_utils
+
+logger = logging.getLogger(__name__)
 
 
 def delete_domains_by_lei(session: Session, lei: str) -> List[FinancialInstitutionDomainDao] | None:
-    session.query(FinancialInstitutionDomainDao).filter(FinancialInstitutionDomainDao.lei == lei).delete()
-    session.commit()
-    return {"OK": True}
+    repo_utils.delete_helper(session, FinancialInstitutionDomainDao, lei)
 
 
 def delete_sbl_type_by_lei(session: Session, lei: str) -> List[SblTypeMappingDao] | None:
-    session.query(SblTypeMappingDao).filter(SblTypeMappingDao.lei == lei).delete()
-    # deleting from history tables
-    del_hist_stmt = text("DELETE from fi_to_type_mapping_history where fi_id = :fi_id")
-    session.execute(del_hist_stmt, {"fi_id": lei})
-    session.commit()
-    return {"OK": True}
+    stmt = session.query(SblTypeMappingDao).filter(SblTypeMappingDao.lei == lei)
+    sbl_types = session.execute(stmt)
+
+    if sbl_types:
+        repo_utils.delete_helper(session, SblTypeMappingDao, lei)
```

Comment:
Same thing here, make sure the delete_helper isn't trying to do SblTypeMappingDao.id = lei because that won't work.